src/backend/cod2.c

```diff
@@ -5108,69 +5104,86 @@ code *cddctor(elem *e,regm_t *pretregs)
 
 code *cdddtor(elem *e,regm_t *pretregs)
 {
-#if MARS
-    /* Generate:
-        ESCAPE | ESCddtor
-        MOV     sindex[BP],index
-        CALL    dtor
-        JMP     L1
-    Ldtor:
-        ... e->E1 ...
-        RET
-    L1: NOP
-    */
-    usednteh |= EHcleanup;
-    if (config.exe == EX_WIN32)
-    {   usednteh |= NTEHcleanup | NTEH_try;
-        nteh_usevars();
-    }
-
-    code cs;
-    cs.Iop = ESCAPE | ESCddtor;
-    cs.Iflags = 0;
-    cs.Irex = 0;
-    cs.IFL1 = FLdtor;
-    cs.IEV1.Vtor = e;
-    code *cd = gen(CNIL,&cs);
+    if (config.ehmethod == EH_DWARF)
+    {
+        usednteh |= EHcleanup;
```

Comment:
> Ok, but this is aimed at adding support for dwarf eh. Re-engineering the entire eh handling semantics is beyond the scope of this.

Sure, just want to make sure that we don't make any mistake and block future work.